shared/instance/instance.go

```diff
@@ -1,4 +1,11 @@
 package instance
 
-// TypeContainer represents a container instance type.
-const TypeContainer = "container"
+// Type indicates the type of instance.
+type Type int
```

Comment:
So that feels wrong. You moved the package from lxd/instance to shared/instance which indicates it contains public API but then you switch it from the API visible string (`container`) to the DB-specific integer.

A container being type == 0 should only matter for the LXD db, nothing external should ever see that.